base/shell/cmd/for.c

```diff
@@ -470,6 +475,7 @@ ExecuteFor(PARSED_COMMAND *Cmd)
     lpNew = cmd_alloc(sizeof(FOR_CONTEXT));
     if (!lpNew)
     {
+        WARN("DEBUG: Cannot allocate memory for lpNew!\n");
         cmd_free(List);
         return 1;
```

Comment:
1?
Maybe TRUE or FALSE?

Reply:
If the function doesn't return BOOL the '1' should be an errorlevel value.

Reply:
This is indeed an errorlevel value: I've changed the return type from BOOL to INT.